src/helpers/backwardCompatibilityForProperties.js

```diff
@@ -1,3 +1,4 @@
-export function backwardCompatibilityForProperties(valuesArrayByMostUpdateNaming) {
-  return valuesArrayByMostUpdateNaming.find(value => value !== undefined);
+export function backwardCompatibilityForProperties(valuesArrayByMostUpdateNaming, defaultValue) {
```

Comment:
1. please add defaults to the arguments `EMPTY_ARRAY`
2. Please consider adding useMemo on the spread of the array which is being passed.